ark/utils/spatial_analysis_utils.py

```diff
@@ -9,30 +9,55 @@
 import os
 
 
-def calc_dist_matrix(label_map, path=None):
+def calc_dist_matrix(label_map, centroid_labels=None, path=None):
     """Generate matrix of distances between center of pairs of cells
 
     Args:
         label_map (np array): array with unique cells given unique pixel labels
+        centroid_labels (dict): the labels for each fov which are
+            needed to add correct coordinates to access distance matrix
+            needs to be of the same length and indexed corresponding to label_map.coords['fovs']
+            if None, then default assume that cell labels are in order for each fov
```

Comment:
I don't think we ever need to provide centroid labels. We haven't created the distance matrix yet. So even if we pass in label_map with non-sequentially labeled cells, that's okay, because we'll record that when we create the xarray of dist_matrices.